core/src/main/scala/kafka/log/LogSegment.scala

```diff
@@ -112,12 +112,13 @@ class LogSegment(val log: FileMessageSet,
    * @param startOffset A lower bound on the first offset to include in the message set we read
    * @param maxSize The maximum number of bytes to include in the message set we read
    * @param maxOffset An optional maximum offset for the message set we read
+   * @param maxPosition An optional maximum position in the log segment that should be exposed for read.
```

Comment:
maxPosition is no longer optional. It would be good to add in the comment that maxPosition is only used when maxOffset is None.